### PR TITLE
Add funcs for gluster backend disk oprations

### PR DESCRIPTION
**Pull request by Yingshun**
This adds some functions to manage the glusterfs backend disk

1. Add functions to mount/umount/check status of glusterfs
2. Add functions to create/remove/check symlink
3. Move setup_or_cleanup_gluster from libvirt.py to gluster.py
    and move get_host_ipv4_addr to utils_net due to cyclic import

This PR is required by migration cases which need to migrate
with gluster backend disk.

Signed-off-by: Yingshun Cui <yicui@redhat.com>

### CODE REVIEW

File: virttest/utils_net.py [hunk 1]
```diff
@@ -3386,26 +3386,41 @@ def get_ip_address_by_interface(ifname, ip_ver="ipv4", linklocal=False):
         return None
 
 
-def get_host_ip_address(params, ip_ver="ipv4", linklocal=False):
+def get_host_ip_address(params=None, ip_ver="ipv4", linklocal=False, ifname=None):
```
Comment (kylazhang) [inline virttest/utils_net.py]: I think no need to add extra parameter ifname, just specify it in param

File: virttest/utils_test/libvirt.py [hunk 2]
```diff
@@ -507,76 +507,6 @@ def setup_or_cleanup_iscsi(is_setup, is_login=True,
...
-        ipv4_value = utils_net.get_net_if_addrs(i)["ipv4"]
-        logging.debug("ipv4_value is %s", ipv4_value)
-        if ipv4_value != []:
-            ip_addr = ipv4_value[0]
-            break
-    if ip_addr is not None:
-        logging.info("ipv4 address is %s", ip_addr)
-    else:
-        raise exceptions.TestFail("Fail to get ip address")
-    return ip_addr
```
Comment (kylazhang) [inline virttest/utils_test/libvirt.py]: Still have other case using this func.
./libvirt/tests/src/virsh_cmd/volume/virsh_volume.py:                    ip_addr = utlv.get_host_ipv4_addr()

Comment (kylazhang) [reply virttest/utils_test/libvirt.py]: why remove it as it can get ip without giving ifname and usable?

File: virttest/utils_misc.py [hunk 3]
```diff
@@ -1511,6 +1511,46 @@ def check_isfile(file_name, session=None):
...
+
+    :param link_name: Link name to be checked
+    :param session: ShellSession object of VM/remote host
+    """
+    if session:
+        cmd = "file %s | grep -q 'symbolic link to'" % link_name
+        return session.cmd_status(cmd) == 0
+    return os.path.islink(link_name)
+
+
```
Comment (kylazhang) [inline virttest/utils_misc.py]: It should be a separate PR?

File: virttest/utils_test/libvirt.py (hunk 2 above)
Comment (Yingshun) [reply virttest/utils_test/libvirt.py]: > Still have other case using this func.
> ./libvirt/tests/src/virsh_cmd/volume/virsh_volume.py: ip_addr = utlv.get_host_ipv4_addr()

It's covered by https://github.com/autotest/tp-libvirt/pull/2281


> why remove it as it can get ip without giving ifname and usable?
It's similar with  get_host_ip_address from functionality in utils_net.py, and it's used by virsh_volume.py in tp-libvirt and original setup_or_cleanup_gluster in avocado-vt.
The func get_host_ip_address can also get ip address without ifname specified -  when it's None, it will get default interface for now.

File: virttest/utils_net.py (hunk 1 above)
Comment (Yingshun) [reply virttest/utils_net.py]: In tp-qemu, they need to check ip_addr from given host_ip_addr or netdst parameter. we have netdst either.
But I want to get ip_address from default or given interface name.
So I add a new param ifname here, and when it's None, it will get ip address from default interface.

File: virttest/utils_misc.py (hunk 3 above)
Comment (Yingshun) [reply virttest/utils_misc.py]: They're required by gluster backend disk feature, so I put them into this PR, you want a separate one for above?

File: virttest/utils_test/libvirt.py (hunk 2 above)
Comment (kylazhang) [reply virttest/utils_test/libvirt.py]: ok, thanks

File: virttest/utils_net.py (hunk 1 above)
Comment (Yingshun) [reply virttest/utils_net.py]: @kylazhang Updated.

File: virttest/gluster.py [hunk 4]
```diff
@@ -236,15 +240,45 @@ def gluster_vol_create(vol_name, hostname, brick_path, force=False, session=None
...
 
-def glusterfs_mount(g_uri, mount_point):
+@error_context.context_aware
+def glusterfs_mount(g_uri, mount_point, session=None):
     """
     Mount gluster volume to mountpoint.
 
     :param g_uri: stripped gluster uri from create_gluster_uri(.., True)
     :type g_uri: str
+    :param mount_point: moun tpoint, str
```
Comment (dzhengfy) [inline virttest/gluster.py]: Correct the typo

File: virttest/gluster.py [hunk 5]
```diff
@@ -422,3 +456,55 @@ def gluster_nfs_disable(vol_name, session=None):
         output = decode_to_text(process.system_output(cmd2))
 
     return 'nfs.disable: on' in output
+
+
+def setup_or_cleanup_gluster(is_setup, vol_name, brick_path="", pool_name="",
+                             file_path="/etc/glusterfs/glusterd.vol", **kwargs):
+    """
+    Set up or clean up glusterfs environment on localhost
```
Comment (dzhengfy) [inline virttest/gluster.py]: Not only on localhost, so please refine the description.

Comment (Yingshun) [reply virttest/gluster.py]: updated

File: virttest/gluster.py (hunk 4 above)
Comment (Yingshun) [reply virttest/gluster.py]: updated.
